### PR TITLE
[REVIEW] Feature/filter operator migration

### LINKED ISSUE #224: Migrate filter operator to new cudf 0.12

### PR DESCRIPTION
**Pull request by jeanp413**
Closes #224

### CODE REVIEW

File: engine/src/CalciteInterpreter.cpp [hunk 1]
```diff
@@ -217,22 +212,22 @@ project_plan_params parse_project_plan(blazing_frame & input, std::string query_
...
-				right_inputs,
-				outputs,
-				operators,
-				unary_operators,
-				left_scalars,
-				right_scalars,
-				new_column_indices,
-				final_output_positions,
-				output.get_gdf_column());
+			// add_expression_to_plan(input,
```
Comment (felipeblazing) [inline engine/src/CalciteInterpreter.cpp]: is this supposed to be commented? whre does this happen now?

File: engine/src/CalciteInterpreter.cpp [hunk 2]
```diff
@@ -644,7 +560,7 @@ blazing_frame evaluate_split_query(std::vector<std::vector<gdf_column_cpp>> inpu
 			return child_frame;
 		} else if(is_filter(query[0])) {
 			blazing_timer.reset();  // doing a reset before to not include other calls to evaluate_split_query
-			process_filter(queryContext, child_frame, query[0]);
+			// process_filter(queryContext, child_frame, query[0]);
```
Comment (felipeblazing) [inline engine/src/CalciteInterpreter.cpp]: why is this commented out? how is it done now?

File: engine/src/Interpreter/interpreter_cpp.cu [hunk 3]
```diff
@@ -216,8 +216,573 @@ void calculate_grid(int * min_grid_size, int * block_size, column_index_type max
...
+//		nv_strings ? NVCategory::create_from_strings(*nv_strings) : NVCategory::create_from_array(nullptr, 0);
+//	NVStrings::destroy(nv_strings);
+
+//	gdf_column_cpp new_input_col;
+//	new_input_col.create_gdf_column(nv_category, nv_category->size(), "");
+
+//	return new_input_col;
+}
+
+int insert_string_into_column_nvcategory(const cudf::column_view & col, const std::string & str) {
```
Comment (felipeblazing) [inline engine/src/Interpreter/interpreter_cpp.cu]: no longe rneed this function

File: engine/src/Interpreter/interpreter_cpp.cu [hunk 4]
```diff
@@ -216,8 +216,573 @@ void calculate_grid(int * min_grid_size, int * block_size, column_index_type max
...
+	static const std::regex any_string_re{R"(([^\\]?|\\{2})%)"};
+	re = std::regex_replace(re, any_string_re, "$1(?:.*?)");
+
+	static const std::regex any_char_re{R"(([^\\]?|\\{2})_)"};
+	re = std::regex_replace(re, any_char_re, "$1(?:.)");
+
+	return (match_start ? "^" : "") + re + (match_end ? "$" : "");
+}
+
+std::unique_ptr<cudf::column> handle_match_regex(const cudf::column_view & input_col, const std::string & re) {
```
Comment (felipeblazing) [inline engine/src/Interpreter/interpreter_cpp.cu]: donde need special sting functions anymore these are going to be handled in interops

File: engine/src/Interpreter/interpreter_cpp.cu [hunk 5]
```diff
@@ -216,8 +216,573 @@ void calculate_grid(int * min_grid_size, int * block_size, column_index_type max
...
+					right_scalars.emplace_back(nullptr);
+				} else if(is_literal(right_operand) && !is_string(right_operand)) {
+					cudf::size_type left_index = get_index(left_operand);
+					auto scalar_ptr = get_scalar_from_string(right_operand);
+					
+					left_inputs.push_back(left_index);
+					right_inputs.push_back(scalar_ptr ? SCALAR_INDEX : SCALAR_NULL_INDEX);
+					left_scalars.emplace_back(nullptr);
+					right_scalars.push_back(std::move(scalar_ptr));
+				} else if(is_string(left_operand) || is_string(right_operand)) {
```
Comment (felipeblazing) [inline engine/src/Interpreter/interpreter_cpp.cu]: dont need special logic here for stirngs anymore

File: engine/src/execution_graph/logic_controllers/LogicalFilter.cpp [hunk 6]
```diff
@@ -1,111 +1,218 @@
...
-    return std::make_unique<ral::frame::BlazingTable>(std::move(
-      filteredTable),table.names());
+  const CudfColumnView & boolValues){
+  auto filteredTable = cudf::experimental::apply_boolean_mask(
+    table.view(),boolValues);
+  return std::make_unique<ral::frame::BlazingTable>(std::move(
+    filteredTable),table.names());
+}
+
+std::unique_ptr<cudf::column> evaluate_expression(
```
Comment (felipeblazing) [inline engine/src/execution_graph/logic_controllers/LogicalFilter.cpp]: this function should not exist. it should be the same function that is used for project and retuns a unique_ptr<BlazingTable>

File: engine/src/CalciteInterpreter.cpp (hunk 1 above)
Comment (jeanp413) [reply engine/src/CalciteInterpreter.cpp]: this is being tackle in #247

File: engine/src/CalciteInterpreter.cpp (hunk 2 above)
Comment (jeanp413) [reply engine/src/CalciteInterpreter.cpp]: same as above

File: engine/src/execution_graph/logic_controllers/LogicalFilter.cpp (hunk 6 above)
Comment (jeanp413) [reply engine/src/execution_graph/logic_controllers/LogicalFilter.cpp]: @rommelDB is working on the logical project operator so maybe this function needs to be deleted in his PR and replaced for the one used in logical project

**Pull request merged**